Months only picker

viorelsfetea opened the pull request:
Uploading code for review before proceeding

Can be instantiated with:

```javascript
       flatpickr(".date", {
          altInput: true,
          defaultDate: new Date('10 february 2019'),
          plugins: [new monthSelect({shorthand: false, dateFormat: "m.y", altFormat: "F Y"})]
        });
```

Features:

- [x] Show months as shorthand or normal name
- [x] Supports locale
- [x] Supports altInput
- [x] Format can be defined both for normal and altInput (format needs to be defined in plugin instantiation and it overwrites flatpickr format definitions) - when no format is defined, it defaults to `F Y`
- [x] Supports themes (needs to be implemented in another commit)
- [ ] Works with confirmDate plugin (works partially, `showAlways: false` currently not working, needs to be implemented in another commit)
- [x] Accessibility 
- [x] Implement `onDestroy`

chmln commented:
@viorelsfetea I will fix up the accessibility part and it's good to go :tada:

viorelsfetea commented:
Great! How should I approach the theming part? Should I just add the styles to each of the styl files for each of the themes?

chmln commented:
@viorelsfetea for now we can do the same thing the `confirmDate` plugin does - just have a light (default) and dark theme.

benjivm commented:
Interested in this functionality, any updates on the status of this PR?

viorelsfetea commented:
@benjivm almost there, just polishing it up

@chmln I added the following: 

1. an onDestroy method clearing the listener added via addEventListener. The ones added with _bind are cleared by the main instance, right?

2. a theme parameter that accepts the `dark` param

3. Renamed the files to monthSelectPlugin because the global variable that's exported for CDN use is named after the file name so people would have had to use "new index()".

benjivm commented:
Thanks a lot @viorelsfetea @chmln !

chmln commented:
You're welcome @benjivm :+1:

spotman commented:
Is there any chance for this PR will be finished? May I help you?

flamendless commented:
How about month only with NO year?

SalkiDixit commented:
@chmln Hi, I have tried using month picker only (monthSelectPlugin), this is coming but styles are not working.
I have imported it
    import * as monthSelectPlugin from "flatpickr/dist/plugins/monthSelect";

I suppose, style should work since i have imported entire plugin.
Please assist.

benjivm commented:
@SalkiDixit No, styles are not automatically imported so you can load your own theme.

```js
import MonthSelectPlugin from 'FlatpickerPlugins/monthSelect';
import 'FlatpickerPlugins/monthSelect/style.css';
```

yogithesymbian commented:
@benjivm 
how about vue2 ? 
```
dpconfig: {
        // default
        // wrap: true,
        // altInput: true,
        // locale: Indonesian,
        // defaultDate: 'today',
        wrap: true,
        enableTime: true,
        time_24hr: true,
        altInput: true,
        altFormat: 'F Y',
        dateFormat: 'Y-m-d',
        // noCalendar: true,
        locale: Indonesian,
        minDate: null,
        maxDate: null,
        defaultDate: null
      },
....
.....
<style lang="scss">
@import '@core/scss/vue/libs/vue-flatpicker.scss';
</style>
```

yogithesymbian commented:
ah i got it on example https://github.com/ankurk91/vue-flatpickr-component/blob/main/examples/App.vue
